### PR TITLE
More Nuts and Trees

### PR DESCRIPTION
**Pull request by cosmiccoincidence**
#### Summary
```SUMMARY: Content "Adds more nuts and trees"```  

#### Purpose of change
- Adds peanuts, pistachios, cashews, almonds, hazelnuts, chestnuts, walnuts, pecans & beech nuts.
- Adds walnut tree, chestnut tree, hazelnut tree, beech tree, pecan tree, almond tree, pistachio tree, peanut bush (nuts without trees can only be found at grocery stores).
- Moved acorns from underbrush forage to oak tree harvest.
- Added imitation peanutbutter and recipes for both peanutbutter and new imitation alternative.

#### Additional informaiton
- Balancing for trees and bushes in forests, thick forests, and swamps is not final. This is only a rough balancing.
- Added nut's to grocery stores' bread item group due to being too lazy to make a new item group and place it. 
- Pecan tree, almond tree, pistachio tree and peanut bush are non-local species and thus will be found only at specific locations like orchards, farms, gardens, etc, not in the wild. 
- Will add plants to specific locations in a later PR.

### CODE REVIEW

**Comment by Marrim** [comment]
IIRC, acorns were removed from oak tree harvest because the oaks growing in New England are not of the sort producing edible acorns.

**Comment by dissociativity** [comment]
As far as I am aware all Acorns technically edible and nutritious, how awful they taste depends upon the type.

**Comment by Mecares** [comment]
Nice that makes Craftable makeshift Nutella a step closer.

**Comment by dissociativity** [comment]
Sifted through the .json, may have missed something but your json doesn't appear to have any syntax errors, so I'm pretty sure it's the checks being picky and a lint should fix it in theory.

**Comment by cosmiccoincidence** [comment]
> IIRC, acorns were removed from oak tree harvest because the oaks growing in New England are not of the sort producing edible acorns.

> EDIT: link to relevant commit below
> 56d4bc6

This seems to be referring to blackjack oaks which I added acorn harvests only to normal oaks.

**Comment by Reclusive-reptile** [comment]
Acorns found by foraging could be squirrel caches.

**Comment by cosmiccoincidence** [comment]
> Acorns found by foraging could be squirrel caches.

Fair point but with oak trees now dropping acorns I don't want to over-populate the world with acorns by having 2 different places you can find them in the same season. Besides, keeping acorns in underbrush would reduce chances of finding other stuff from foraging.

Although maybe having acorns foragable from underbrush either in winter or spring, AFTER squirrels had the chance to stash many away and they are not available via trees during those seasons.

**Comment by cosmiccoincidence** [comment]
Notice me, senpai...

@ZhilkinSerg

**Comment by dissociativity** [comment]
I'm waiting to push my PK Rebalancing update for this :P
Some pls merge thanks

File: data/json/items/comestibles.json [hunk 1]
```diff
@@ -4324,44 +4324,335 @@
...
+    "name_plural": "handfuls of roasted pecans",
+    "weight": 36,
+    "color": "brown",
+    "spoils_in": 8640,
+    "comestible_type": "FOOD",
+    "symbol": "%",
+    "quench": -3,
+    "healthy": 1,
+    "calories": 120,
+    "description": "A handful of roasted nuts from a walnut tree.",
```
Comment (Cyrano7) [inline data/json/items/comestibles.json]: Shouldn't this be a "pecan" tree?

File: data/json/items/comestibles.json [hunk 2]
```diff
@@ -4324,44 +4324,335 @@
...
+    "weight": 32,
+    "color": "brown",
+    "spoils_in": 8640,
+    "container": "bag_plastic",
+    "comestible_type": "FOOD",
+    "symbol": "%",
+    "quench": -3,
+    "healthy": 2,
+    "calories": 80,
+    "description": "An imported bag of almonds.",
```
Comment (Cyrano7) [inline data/json/items/comestibles.json]: You don't have to say they are imported. America is the number one producer of almonds in the world.

File: data/json/items/comestibles.json (hunk 1 above)
Comment (cosmiccoincidence) [reply data/json/items/comestibles.json]: Correct.

File: data/json/items/comestibles.json (hunk 2 above)
Comment (cosmiccoincidence) [reply data/json/items/comestibles.json]: Adjustment made, as well as added tree for almonds for use in orchards.

**Comment by cosmiccoincidence** [comment]
Changes made.

**Comment by ZhilkinSerg** [comment]
Please review and lower shelf life of all nuts. First random link shows 1-2 years in freezer, not 6 years at room temperature - http://www.eatbydate.com/proteins/nuts/how-long-do-nuts-last-shelf-life-expiration-date/

**Comment by cosmiccoincidence** [comment]
I don't have anything at a 6 year spoil time do I? I thought they were all set to 1 year. But ill look into balancing them some more...

**Comment by cosmiccoincidence** [comment]
Made nut have the following spoil times based on Serg's link...
pine nuts = 1.5 months
pistachios = 3 months
beech nuts = 3 months
hazelnuts = 5 months
hickory nuts = 6 months
pecans = 6 months
walnuts = 6 months
chestnuts = 6 months
cashews = 8 months
peanuts = 8 months
acorns = 8 months
almonds = 10 months

File: data/json/items/comestibles.json [hunk 3]
```diff
@@ -4324,44 +4324,355 @@
...
+    "weight": 28,
+    "color": "green",
+    "spoils_in": 2160,
+    "container": "bag_plastic",
+    "comestible_type": "FOOD",
+    "symbol": "%",
+    "quench": -3,
+    "healthy": 2,
+    "calories": 80,
+    "description": "An imported bag of salty pistachios, their shells have been removed.",
```
Comment (Cyrano7) [inline data/json/items/comestibles.json]: America is also a world leader in pistachios production.

Comment (cosmiccoincidence) [reply data/json/items/comestibles.json]: Added pistachio tree.

**Pull request merged**